Sources/CCFilter/ViewControllers/CCFilterViewController.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import UIKit
+
+public protocol CCFilterViewControllerDelegate: class {
+    func filterViewControllerFilterSelectionChanged(_ filterViewController: CCFilterViewController)
+    func filterViewControllerDidPressShowResults(_ filterViewController: CCFilterViewController)
+    func filterViewController(_ filterViewController: CCFilterViewController, didSelectVerticalAt index: Int)
+}
+
+public class CCFilterViewController: UINavigationController {
+
+    // MARK: - Public properties
+
+    public var filter: CCFilter {
+        didSet {
+            rootFilterViewController.set(filterNode: filter.root, verticals: filter.verticals)
+            isLoading = false
+        }
+    }
+
+    public var isLoading: Bool = false {
+        didSet {
+            updateLoading()
+        }
+    }
+
+    public var config: CCFilterConfiguration
+    public weak var filterDelegate: CCFilterViewControllerDelegate?
+
+    public var mapFilterViewManager: MapFilterViewManager?
+    public var searchLocationDataSource: SearchLocationDataSource?
+
+    // MARK: - Private properties
+
+    private let selectionStore: FilterSelectionStore
+    private var rootFilterViewController: CCRootFilterViewController
+
+    private lazy var loadingViewController = LoadingViewController(backgroundColor: .milk, presentationDelay: 0)
+
+    // MARK: - Init
+
+    public init(filter: CCFilter, config: CCFilterConfiguration) {
+        self.filter = filter
+        self.config = config
+        selectionStore = FilterSelectionStore()
+        rootFilterViewController = CCRootFilterViewController(filterNode: filter.root, selectionStore: selectionStore)
+        rootFilterViewController.verticals = filter.verticals
+        super.init(nibName: nil, bundle: nil)
+        rootFilterViewController.rootDelegate = self
+        setViewControllers([rootFilterViewController], animated: false)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Lifecycle
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        delegate = self
+    }
+
+    // MARK: - Private
+
+    private func updateLoading() {
+        if isLoading {
+            add(loadingViewController)
+            loadingViewController.viewWillAppear(false)
+        } else {
+            loadingViewController.remove()
+        }
+    }
+}
+
+// MARK: - CCRootFilterViewControllerDelegate
+
+extension CCFilterViewController: CCRootFilterViewControllerDelegate {
+    func rootFilterViewController(_ viewController: CCRootFilterViewController, didSelectVerticalAt index: Int) {
+        filterDelegate?.filterViewController(self, didSelectVerticalAt: index)
```

Comment:
It maybe makes sense to tell the delegate which vertical that was selected, the index is probably a lot less useful?